force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -26,184 +26,164 @@
-
-        return null; // Replace null with the variable you used to store the result
+        return result; 
     }
 
     /**
-     * Question 5
      * Given a list of scientist names, combine their names together with a comma if it contains the letter 'g'.
      * The final string should not end with a comma.
      * Example: findTheScientist() should return "Alan Turing,Grace Hopper,Guido van Rossum,Stephen Hawking"
      * @return A string containing the names of scientists that contain the letter 'g'.
      */
     public static String findTheScientist() {
-        // The list of scientists' names
         List<String> scientistNames = new List<String>{'Tim Berners-Lee', 'Alan Turing', 'Grace Hopper', 'Donald Knuth', 'Guido van Rossum', 'Ken Thompson', 'Stephen Hawking'};
 
-        // The variable to store the concatenated string
-
-        // Loop through the list of scientists' names
+        String combinedNames = '';
 
-            // If the name doesn't contain the letter 'G', skip this iteration
-
-            // Add the name to the result string, followed by a comma
-
-        // Remove the last comma from the result string
+        for (String name : scientistNames) {
+            String lowerName = name.toLowerCase();
+            if(!lowerName.contains('g')) {
+                continue;
+                } else {
+                    combinedNames += name + ',';
+                } 
+        }
 
-        return null; // Replace null with the variable you used to store the result
+        combinedNames = combinedNames.removeEnd(',');
+        return combinedNames;
     }
 
     /**
-     * Question 6
      * This method calculates the absolute value of the difference between of two input numbers that are raised to the 2nd power.
      * Example: calculateDifferenceOfSquares(3, 4) should return 7.
      * @param a The first number.
      * @param b The second number.
      * @return The absolute value of the difference
      */
     public static Double calculateDifferenceOfSquares(Double a, Double b) {
-        // Square the input numbers using the Math.pow() function
-
-        // Calculate the absolute difference using Math.abs() function
-
-        // Return the result
-        return null; // Replace null with the variable you used to store the result
+        Double doubleA = math.pow(a, 2);
+        Double doubleB = math.pow(b, 2);
+        Double result = math.abs(doubleA - doubleB);
+        return result;
     }
 
     /**
-     * Question 7
      * Generate a random number between 50 and 100
      * Example: generateRandomNumber() should return a number like 75.
      * @return A integer between 50 and 100.
      */
     public static Integer generateRandomNumber() {
-        // Use Math class to generate number between 50 and 100
-
-        // Return the random integer
-        return null; // Replace null with the variable you used to store the result
+        Integer randomNumber = (Integer)Math.random() + 51;
```

Comment:
If you were to put this inside a loop and run it 10 times (with a `System.debug()` line), what results would you expect to see? 

Here's what happens:
```
12:47:04.24 (25183416)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25226563)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25252393)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25275493)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25313038)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25337084)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25360937)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25383899)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25406999)|USER_DEBUG|[3]|DEBUG|random: 51
12:47:04.24 (25430255)|USER_DEBUG|[3]|DEBUG|random: 51
```

How might you fix it? Here's a template that should help:
`(Integer) (Math.random() * (max - min + 1)) + min`